File: TsinghuaNetHelper/LiveTileTask.cpp
#include "pch.h"

#include "LiveTileTask.h"

using namespace winrt;
using namespace Windows::ApplicationModel::Background;

namespace winrt::TsinghuaNetHelper::implementation
{
    fire_and_forget LiveTileTask::Run(IBackgroundTaskInstance const taskInstance)
    {
        auto deferral{ taskInstance.GetDeferral() };
        try
        {
            hstring ssid;
            auto status{ SettingsHelper::InternetStatus(ssid) };
            NetState state{ settings.SuggestNetState(status, ssid) };
            IConnect helper{ ConnectHelper::GetHelper(state) };
            if (helper)
            {
                FluxUser user{ co_await helper.FluxAsync() };
                NotificationHelper::UpdateTile(user);
            }
        }
        catch (hresult_error const&)
        {
        }
        deferral.Complete();
    }
} // namespace winrt::TsinghuaNetHelper::implementation

File: TsinghuaNetUWP.Background/LiveTileTask.vb
Imports Berrysoft.Tsinghua.Net
Imports TsinghuaNetUWP.Helper
Imports Windows.ApplicationModel.Background

Public NotInheritable Class LiveTileTask
    Implements IBackgroundTask

    Public Async Sub Run(taskInstance As IBackgroundTaskInstance) Implements IBackgroundTask.Run
        Dim deferral = taskInstance.GetDeferral()
        Try
            Dim tuple = SettingsHelper.GetInternetStatus()
            Dim state = SettingsHelper.SuggestNetState(tuple.Status, tuple.Ssid)
            Dim helper = ConnectHelper.GetHelper(state)
            If helper IsNot Nothing Then
                Using helper
                    Dim user As FluxUser = Await helper.GetFluxAsync()
                    NotificationHelper.UpdateTile(user)
                End Using
            End If
        Finally
            deferral.Complete()
        End Try
    End Sub
End Class
